[BUG] go-raml on Windows uses wrong separator for generating resource strings

How to replicate on Windows:
1. Create a simple RAML file with one resource, eg. `/SomeResource`
2. Run `go-raml server ...` on the RAML file.

Expected:
Generation completes successfully

Actual:
Generation cuts off due to error:
```
time="2016-11-18T16:47:43+08:00" level=error msg="Error running go fmt on './server/<someresource>_if.go' failed:\n./server/someresource_if.go:20:18: unknown escape sequence\n"
```

Observation:
Inside the code generation, resource-URI-forming uses golang's default `filepath.Join()` method, which uses the default file separator of the operating system. This means that on Windows, the URI will be formed with backslash (`\`), which is not the correct separator for REST-ful URIs. Because of this, using `go-raml` fails on Windows.

This line appears to be the culprit: https://github.com/Jumpscale/go-raml/blob/master/raml/resource.go#L312

Suggestion:
I recommend refactoring all functions that handle REST-ful URIs to use a new function that wraps `strings.Join( x, '/')` explicitly instead. Specifying the separator explicitly will make it properly cross-platform

--- a/raml/resource.go
+++ b/raml/resource.go
@@ -3,7 +3,7 @@ package raml
 import (
 	"fmt"
 	"log"
-	"path/filepath"
+	"path"
 	"strings"
 )
 
@@ -309,7 +309,7 @@ func (r *Resource) FullURI() string {
 }
 
 func doFullURI(r *Resource, completeURI string) string {
-	completeURI = filepath.Join(r.URI, completeURI)
+	completeURI = path.Join(r.URI, completeURI, "/")
 	if r.Parent == nil {
 		return completeURI
 	}

